cpp/stdlib.cc

```diff
@@ -120,6 +142,85 @@ void kill(int pid, int sig) {
   }
 }
 
+List<Str*>* listdir(Str* path) {
+  DIR* dirp = opendir(path->data());
+  if (dirp == NULL) {
+    throw Alloc<OSError>(errno);
+  }
+
+  auto* ret = NewList<Str*>();
+  while (true) {
+    errno = 0;
+    struct dirent* ep = readdir(dirp);
+    if (ep == NULL) {
+      if (errno == 0) {
+        break;
+      } else {
+        closedir(dirp);
+        throw Alloc<OSError>(errno);
+      }
+    }
+    if (ep->d_name[0] == '.' &&
+        (NAMLEN(ep) == 1 || (ep->d_name[1] == '.' && NAMLEN(ep) == 2))) {
+      continue;
+    }
+    ret->append(StrFromC(ep->d_name, NAMLEN(ep)));
+  }
+
+  closedir(dirp);
+
+  return ret;
+}
+
+stat_result::stat_result(const struct stat* st)
+    : GC_CLASS_FIXED(header_, kZeroMask, sizeof(stat_result)) {
+  n_fields = 0;
+  n_sequence_fields = 0;
+  n_unnamed_fields = 0;
+
+  st_mode = st->st_mode;
+  st_ino = st->st_ino;
+  st_dev = st->st_dev;
+  st_nlink = st->st_nlink;
+
+#if defined(MS_WINDOWS)
+  st_uid = st->st_uid;
+  st_gid = st->st_gid;
+#else
+  st_uid = 0;
+  st_gid = 0;
+#endif
+
+  st_size = st->st_size;
+
+#if defined(HAVE_STAT_TV_NSEC)
```

Comment:
Ditto here, I don't know this interface, but I think we should be able to do it without `#ifdef`

All we are doing is testing the mtime of a directory?  That should be doable in portable POSIX?  The resolution doesn't really matter AFAICT


I guess if this unblocks us, it could be worth putting in, but I would want to make another pass over it